server/go/pkg/server/server.go

```diff
@@ -0,0 +1,295 @@
+	"github.com/godaddy/asherah/go/appencryption/pkg/crypto/aead"
+	"github.com/godaddy/asherah/go/appencryption/pkg/kms"
+	"github.com/godaddy/asherah/go/appencryption/pkg/persistence"
+	"github.com/godaddy/asherah/go/securememory/memguard"
+
+	pb "github.com/godaddy/asherah/server/go/api"
+)
+
+var (
+	UnitializedSessionResponse        = newErrorResponse("session not yet initialized")
+	SessionAlreadyInitializedResponse = newErrorResponse("session has already been initialized")
+)
+
+func newErrorResponse(message string) *pb.SessionResponse {
+	return &pb.SessionResponse{
+		Response: &pb.SessionResponse_ErrorResponse{
+			ErrorResponse: &pb.ErrorResponse{Message: message},
+		},
+	}
+}
+
+type requestHandler interface {
+	Decrypt(context.Context, *pb.SessionRequest) *pb.SessionResponse
+	Encrypt(context.Context, *pb.SessionRequest) *pb.SessionResponse
+	GetSession(*pb.SessionRequest) *pb.SessionResponse
+	Close() error
+}
+
+type AppEncryption struct {
+	pb.UnimplementedAppEncryptionServer
+	streamerFactory
+}
+
+type streamerFactory interface {
+	NewStreamer() *streamer
+}
+
+type streamerFactoryFunc func() *streamer
+
+func (s streamerFactoryFunc) NewStreamer() *streamer {
+	return s()
+}
+
+func (a *AppEncryption) Session(stream pb.AppEncryption_SessionServer) error {
+	s := a.NewStreamer()
+
+	return s.Stream(stream)
+}
+
+type streamer struct {
+	handlerFactory
+	handler requestHandler
+	options *Options
+}
+
+type handlerFactory interface {
+	NewHandler() requestHandler
+}
+
+func (s *streamer) NewHandler() requestHandler {
+	if s.handlerFactory != nil {
+		return s.handlerFactory.NewHandler()
+	}
+
+	crypto := aead.NewAES256GCM()
+
+	sf := appencryption.NewSessionFactory(
+		&appencryption.Config{
+			Service: s.options.ServiceName,
+			Product: s.options.ProductID,
+			Policy: appencryption.NewCryptoPolicy(
+				appencryption.WithExpireAfterDuration(s.options.ExpireAfter),
+				appencryption.WithRevokeCheckInterval(s.options.CheckInterval),
+			),
+		},
+		NewMetastore(s.options),
+		NewKMS(s.options, crypto),
+		crypto,
+		appencryption.WithSecretFactory(new(memguard.SecretFactory)),
+		appencryption.WithMetrics(false),
+	)
+
+	return &defaultHandler{
+		sessionFactory: sf,
+	}
+}
+
+func NewMetastore(opts *Options) appencryption.Metastore {
+	switch opts.Metastore {
+	case "rdbms":
+		// TODO: support other databases
+		db, err := newMysql(opts.ConnectionString)
+		if err != nil {
+			panic(err)
+		}
+
+		return persistence.NewSQLMetastore(db)
+	case "dynamodb":
+		sess := awssession.Must(awssession.NewSessionWithOptions(awssession.Options{
+			SharedConfigState: awssession.SharedConfigEnable,
+		}))
+
+		return persistence.NewDynamoDBMetastore(sess)
+	default:
+		return persistence.NewMemoryMetastore()
+	}
+}
+
+func NewKMS(opts *Options, crypto appencryption.AEAD) appencryption.KeyManagementService {
+	if opts.KMS == "static" {
+		kms, err := kms.NewStatic("thisistotallynotsecretdonotuse!!", aead.NewAES256GCM())
```

Comment:
variable name `kms` same as package name. Is it fine? I see the IDE complain about it